### PR TITLE
[SettingsBundle][UserBundle] introduce new bundles

### PR DESCRIPTION
**Pull request by ahilles107**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             |no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-613 SWP-581
| License                   | AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/CoreBundle/Resources/config/app/config.yml [hunk 1]
```diff
@@ -192,15 +195,4 @@ swp_user:
...
-        my_second_setting:
-            value: "test"
-            scope: "tenant"
-        my_third_setting: ~
-        array_setting:
-            value: '{"a":"b","c":"d"}'
-            type: array
-            scope: global
\ No newline at end of file
+            enabled: true
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/Resources/config/app/config.yml]: new line is missing

File: src/SWP/Bundle/UserBundle/Mailer/Mailer.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        $this->router = $router;
+        $this->templating = $templating;
+        $this->parameters = $parameters;
+
+        $this->parameters['confirmation.template'] = $settingsManager->get('registration_confirmation.template', 'tenant');
+        $this->parameters['from_email']['confirmation'] = $settingsManager->get('registration_from_email.confirmation', 'tenant');
+        $this->parameters['resetting.template'] = $settingsManager->get('registration_resetting.template', 'tenant');
+        $this->parameters['from_email']['resetting'] = $settingsManager->get('registration_from_email.resetting', 'tenant');
+
+        dump($this->parameters);
```
Comment (takeit) [inline src/SWP/Bundle/UserBundle/Mailer/Mailer.php]: dont forget to remove it

File: src/SWP/Bundle/UserBundle/Resources/config/app/config.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            scope: tenant
+            type: array
+        registration_resetting.template:
+            value: %fos_user.resetting.email.template%
+            scope: tenant
+            type: string
+        registration_from_email.resetting:
+            value: %fos_user.resetting.email.from_email%
+            scope: tenant
+            type: array
```
Comment (takeit) [inline src/SWP/Bundle/UserBundle/Resources/config/app/config.yml]: new line is missing

File: src/SWP/Bundle/UserBundle/Resources/config/routing.yml [hunk 4]
```diff
@@ -1,3 +1,7 @@
 swp_user:
     resource: "@SWPUserBundle/Controller"
-    type:     annotation
\ No newline at end of file
+    type:     annotation
+
+fos_user_register:
+    resource: "@FOSUserBundle/Resources/config/routing/registration.xml"
+    prefix: /register
```
Comment (takeit) [inline src/SWP/Bundle/UserBundle/Resources/config/routing.yml]: new line is missing

so there is also an option to register/login via normal Symfony form?

File: src/SWP/Bundle/UserBundle/Mailer/Mailer.php (hunk 2 above)
Comment (ahilles107) [inline src/SWP/Bundle/UserBundle/Mailer/Mailer.php]: yeah, i will remove it when i will add support for tenant scope to settings ;)

File: src/SWP/Bundle/UserBundle/Resources/config/app/config.yml (hunk 3 above)
Comment (ahilles107) [inline src/SWP/Bundle/UserBundle/Resources/config/app/config.yml]: i will review new lines and debug lines at end ;)

File: src/SWP/Bundle/UserBundle/Resources/config/routing.yml (hunk 4 above)
Comment (ahilles107) [inline src/SWP/Bundle/UserBundle/Resources/config/routing.yml]: God point, i will remove it and replace wit our custom route. We need only register with API.

File: src/SWP/Bundle/SettingsBundle/Manager/SettingsManager.php [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    }
+
+    /**
+     * @param string                      $name
+     * @param string                      $scope
+     * @param SettingsOwnerInterface|null $owner
+     *
+     * @return mixed
+     */
+    private function getSettingFromRepository(string $name, string $scope, SettingsOwnerInterface $owner = null)
```
Comment (takeit) [inline src/SWP/Bundle/SettingsBundle/Manager/SettingsManager.php]: this should go to SettingsRepository

File: src/SWP/Bundle/SettingsBundle/Tests/Functional/app/SettingsBundle/bundles.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+    new Symfony\Bundle\SecurityBundle\SecurityBundle(),
+    new Sensio\Bundle\FrameworkExtraBundle\SensioFrameworkExtraBundle(),
+    new Doctrine\Bundle\DoctrineBundle\DoctrineBundle(),
+    new Doctrine\Bundle\DoctrineCacheBundle\DoctrineCacheBundle(),
+    new Symfony\Bundle\MonologBundle\MonologBundle(),
+    new Liip\FunctionalTestBundle\LiipFunctionalTestBundle(),
+    new Symfony\Bundle\TwigBundle\TwigBundle(),
+    new Nelmio\ApiDocBundle\NelmioApiDocBundle(),
+    //new Stof\DoctrineExtensionsBundle\StofDoctrineExtensionsBundle(),
```
Comment (takeit) [inline src/SWP/Bundle/SettingsBundle/Tests/Functional/app/SettingsBundle/bundles.php]: why is it commented?

File: src/SWP/Bundle/SettingsBundle/Tests/Functional/app/SettingsBundle/config.yml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        main_cache:
+            type: array
+            namespace: settings_bundle
+
+liip_functional_test: ~
+
+sensio_framework_extra:
+    view:    { annotations: false }
+
+#stof_doctrine_extensions:
```
Comment (takeit) [inline src/SWP/Bundle/SettingsBundle/Tests/Functional/app/SettingsBundle/config.yml]: why is it commented?

**Comment by takeit** [comment]
check also scrutinizer issues

File: src/SWP/Bundle/CoreBundle/Tests/Controller/RegistrationController.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+        // Asserting email data
+        $this->assertInstanceOf('Swift_Message', $message);
+        $this->assertEquals('Welcome sofab.contact!', $message->getSubject());
+        $this->assertEquals('webmaster@example.com', key($message->getFrom()));
+        $this->assertEquals('contact@example.com', key($message->getTo()));
+
+        preg_match_all('#\bhttps?://[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|/))#', $message->getBody(), $match);
+
+        $client->enableProfiler();
```
Comment (ahilles107) [inline src/SWP/Bundle/CoreBundle/Tests/Controller/RegistrationController.php]: nope, as it allows us to get details of send email. Later we use code from profiler in assertions.

File: src/SWP/Bundle/SettingsBundle/Tests/Functional/app/SettingsBundle/bundles.php (hunk 6 above)
Comment (ahilles107) [inline src/SWP/Bundle/SettingsBundle/Tests/Functional/app/SettingsBundle/bundles.php]: it's not finished - good catch

File: src/SWP/Bundle/CoreBundle/Tests/Controller/RegistrationController.php [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        self::bootKernel();
+        $this->initDatabase();
+        $this->loadCustomFixtures(['tenant']);
+        $this->router = $this->getContainer()->get('router');
+    }
+
+    public function testRevisionPublishing()
+    {
+        $client = static::createClient();
+        $client->enableProfiler();
```
Comment (ahilles107) [inline src/SWP/Bundle/CoreBundle/Tests/Controller/RegistrationController.php]: nope, as it allows us to get details of send email. Later we use code from profiler in assertions.

File: src/SWP/Bundle/SettingsBundle/Manager/SettingsManager.php (hunk 5 above)
Comment (ahilles107) [inline src/SWP/Bundle/SettingsBundle/Manager/SettingsManager.php]: `findOneByNameAndScopeAndOwner` is already in repository and returns query builder - here we get query and decide what kind of result we want to get.

**Pull request merged**